File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    crown = []
    
    for i in score:
        if len(crown) < k:
            crown.append(i)
        else:
            min_crown = min(crown)
            if i > min_crown:
                crown.remove(min_crown)
                crown.append(i)
        answer.append(min(crown))
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int n = score.length;
        int[] answer = new int[n];
        PriorityQueue<Integer> hall = new PriorityQueue<>();
        
        for (int i = 0; i < n; i++) {
            hall.add(score[i]);
            
            if (i >= k) {
                hall.poll();

            }
            
            answer[i] = hall.peek();
        }
                    
        
        return answer;
    }
}